src/functional/iterator/runtime.py

```diff
@@ -12,10 +12,12 @@
 @dataclass(frozen=True)
 class Offset:
     value: Optional[Union[int, str]] = None
+    source: Optional[common.Dimension] = None
+    target: Optional[Tuple[common.Dimension, common.Dimension]] = None
```

Comment:
Is there any reason why this needs to be a pair or can it be a unrestricted tuple?
```suggestion
    target: Optional[Tuple[common.Dimension, ...] = None
```

Reply:
Works just as well with an unrestricted tuple.

Reply:
Works just as well with an unrestricted tuple.